src/consensus/validator.rs

```diff
@@ -187,12 +201,25 @@ impl ShardValidator {
         round: Round,
         timeout: Duration,
     ) -> FullProposal {
-        if let Some(block_proposer) = &mut self.block_proposer {
+        match self.proposal_store.get_proposal(height, round) {
```

Comment:
It's a bit annoying we have to do this for every propose. I think we only need to do this the first time we restart? But not a huge deal, we can optimize later.

Reply:
Yeah, agreed. Will leave as-is for now and we can optimize later.